### DISCUSSION TITLE
Is Inversify compatible with a pure JavaScript project?

### DISCUSSION THREAD

**Discussion by talkohavy** [inversify/InversifyJS#1533] [Q&A] [answered]
I'm using a pure JavaScript project with a jsconfig.json file.
I have .d.ts file just for types.
My project contains Jsdocs that use the types declared within the .d.ts files.
My entire project is "type: module" (import & export syntax) to keep it as native as possible.

Due to the nature of my project, it does not bundle (no need for babel transpilation, or tsc).
With the above configuration in mind,
is `inversify` applicable to my project?

If the answer is yes, i'd love to see an example ❤️

**Comment by rsaz** [comment, marked as answer]
Hey @talkohavy ,

Yes, any TypeScript package, including Inversify, can indeed run in JavaScript. While TypeScript offers additional capabilities that JavaScript lacks (like decorators), Inversify has you covered!

The Inversify team anticipated this need, so they included a helper function called decorate that makes it easy to apply decorators in JavaScript.

Here’s a simple example to get you started:

`main.js`

```js
import "reflect-metadata";

import { Container, injectable, decorate } from "inversify";

class Warrior {
    fight() {
        return "Warrior fighting";
    }
}

// Manually decorate the Warrior class as injectable
decorate(injectable(), Warrior);

const container = new Container();
container.bind(Warrior).toSelf();

const warrior = container.get(Warrior);
const resp = warrior.fight();
console.log(resp); // Warrior fighting
```

`package.json`
{
    "name": "inversifyjs-demo",
    "version": "1.0.0",
    "main": "main.js",
    "type": "module",
    "scripts": {
        "test": "echo \"Error: no test specified\" && exit 1"
    },
    "keywords": [],
    "author": "",
    "license": "ISC",
    "description": "",
    "dependencies": {
        "inversify": "^6.0.3",
        "reflect-metadata": "^0.2.2"
    }
}

`output`
![image](https://github.com/user-attachments/assets/8099e2ff-727a-4f5d-97e4-d755c1d114ed)

cc: @notaphplover
